src/wms/src/main/java/org/geoserver/wms/MapLayerInfo.java

```diff
@@ -79,16 +79,31 @@
     /** The extra constraints that can be set when an external SLD is used */
     private FeatureTypeConstraint[] layerFeatureConstraints;
 
+    private MetadataMap overrideMetadata;
```

Comment:
This name is going to cause confusion, what you have here is not a generic override. See for example this snipped from line 370:

```java
    public static String getRegionateAttribute(FeatureTypeInfo layerInfo) {
        return layerInfo.getMetadata().get("kml.regionateAttribute", String.class);
    }
```

Here the metadata map is accessed directly. Should we use the "override"? No, we don't really want to, but the name suggests otherwise, the code has been given a metadata map override, why ignore it here.

Better to call it for what it is, a ``layerGroupMetadata``, and allow code to use it on a case by case basis.

Reply:
ok